internal/pkg/utils/utils_test.go

```diff
@@ -544,3 +544,61 @@ func TestBase64Bytes_MarshalYAML(t *testing.T) {
 		})
 	}
 }
+func TestGetSliceFromPointer(t *testing.T) {
+	tests := []struct {
+		name      string
+		input     *[]string
+		expected  []string
+		expectNil bool
+	}{
+		{
+			name:      "nil pointer",
+			input:     nil,
+			expected:  []string{},
+			expectNil: false,
+		},
+		{
+			name: "pointer to nil slice",
+			input: func() *[]string {
+				var s []string
+				return &s
+			}(),
+			expected:  nil,
+			expectNil: true,
```

Comment:
Nice edge case, but the expected output of the utils function should never be nil. Instead an empty slice should be returned
```suggestion
			name: "pointer to nil slice",
			input: func() *[]string {
				var s []string
				return &s
			}(),
			expected:  []string{},
			expectNil: true,
```
To make this test case successful, you need to extend the utils function:
```go
func GetSliceFromPointer[T any](s *[]T) []T {
	if s == nil || *s == nil {
		return []T{}
	}
	return *s
}
```

With this you can probably also remove the attribute `expectNil`